cmd/lachesis/commands/run.go

```diff
@@ -121,6 +122,9 @@ func AddRunFlags(cmd *cobra.Command) {
 	cmd.Flags().String("datadir", config.Lachesis.DataDir, "Top-level directory for configuration and data")
 	cmd.Flags().String("log", config.Lachesis.LogLevel, "debug, info, warn, error, fatal, panic")
 	cmd.Flags().Bool("log2file", config.Log2file, "duplicate log output into file lachesis_<BindAddr>.log")
+	if runtime.GOOS != "windows" {
```

Comment:
You haven't to check "GOOS", use os.TempDir() instead.

Reply:
?! Please explain.

Reply:
https://golang.org/pkg/io/ioutil/#TempDir

Reply:
But it should not work on Windows? You can put the tag `// +build !windows `

Reply:
`
const DefaultPidFileName = filepath.Join(os.TempDir(), "go-lachesis.pid")

func NewDefaultCLIConfig() *CLIConfig {
	return &CLIConfig{
		. . .
		Pidfile:    DefaultPidFileName,
	}
}
. . .
cmd.Flags().String("pidfile", config.Pidfile, "pidfile location; "+DefaultPidFileName+" by default")
`
, then remove `runtime.GOOS != "windows"` conditions

Reply:
First of all pidfile is completely useless on Windows, so no point to create it there.
Secondly, build tags affects whole file, so it is an overkill to use this feature for so small platform difference; plus build tags are not honoured in this projects - i have added a couple for debugging and testing but no one bothers to check if their commit break it.

Using os.TempDir for default location of pidfile make sense, agree.

Reply:
I don't agree pidfile is completely useless on Windows. For example, we can use the same linux-shell-scrypts on Windows as on Linux.
Anyway, cmd/lachesis/commands/run_lachesis.go:runLachesis() must have the same behavior on both OS. And there are no reasons to make an efforts (i.e. GOOS checks) for special behavior on Windows.

But it may make sense to empty "pidfile" default value. No pidfile by default.

Reply:
Pidfiles are used to manage background execution of a program running on unix-like OS under systemd or alike using signals. In Windows signals are rudimentary and Windows Services are used to control background execution, which has its own API. From that perspective pidfiles are useless on Windows.